devsetup/scripts/edpm-compute-bmaas.sh

```diff
@@ -0,0 +1,245 @@
+#!/bin/bash
+#
+# Copyright 2023 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+set -ex
+RHEL_IMAGE_URL=${RHEL_IMAGE_URL:-"https://images.rdoproject.org/centos9/master/rdo_trunk/current-tripleo/overcloud-hardened-uefi-full.qcow2"}
+AGENT_IMAGE_URL=${AGENT_IMAGE_URL:-"quay.io/openstack-k8s-operators/openstack-baremetal-operator-agent:latest"}
+DOWNLOADER_IMAGE_URL=${DOWNLOADER_IMAGE_URL:-"quay.io/openstack-k8s-operators/openstack-baremetal-operator-downloader:latest"}
+APACHE_IMAGE_URL=${APACHE_IMAGE_URL:-"registry.redhat.io/rhel8/httpd-24:latest"}
+NETWORK_NAME=${NETWORK_NAME:-"crc-bmaas"}
+NODE_NAME_PREFIX=${NODE_NAME_PREFIX:-"crc-bmaas"}
+OUTPUT_DIR=${OUTPUT_DIR:-"../out/edpm"}
+NODE_COUNT=${BMAAS_NODE_COUNT:-2}
+NETWORK_IPADDRESS=${BMAAS_NETWORK_IPADDRESS:-172.22.0.3}
+NODE_INDEX=0
+while IFS= read -r instance; do
+    export uuid_${NODE_INDEX}="${instance% *}"
+    name="${instance#* }"
+    export mac_address_${NODE_INDEX}=$(virsh --connect=qemu:///system domiflist "$name" | grep "${NETWORK_NAME}" | awk '{print $5}')
+    echo ${mac_address_0}
+    NODE_INDEX=$((NODE_INDEX+1))
+done <<< "$(virsh --connect=qemu:///system list --all --uuid --name | grep "${NODE_NAME_PREFIX}")"
+
+mkdir -p ${OUTPUT_DIR}
+
+for (( i=0; i<${NODE_COUNT}; i++ )); do
+    mac_var=mac_address_${i}
+    uuid_var=uuid_${i}
+    cat <<EOF >>${OUTPUT_DIR}/bmh_deploy.yaml
+---
+# This is the secret with the BMC credentials (Redfish in this case).
+apiVersion: v1
+kind: Secret
+metadata:
+  name: node-${i}-bmc-secret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: cGFzc3dvcmQ=
+
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: crc-bmaas-${i}
+  annotations:
+    inspect.metal3.io: disabled
+spec:
+  bmc:
+    address: redfish+http://sushy-emulator.apps-crc.testing/redfish/v1/Systems/${!uuid_var}
+    credentialsName: node-1-bmc-secret
+  bootMACAddress: ${!mac_var}
+  bootMode: UEFI
+  online: false
+  rootDeviceHints:
+    deviceName: /dev/vda
+EOF
+done
+
+# Wait for CRDs
+oc wait --for condition=established crd openstackprovisionservers.baremetal.openstack.org
+oc wait --for condition=established crd openstackdataplanes.dataplane.openstack.org
```

Comment:
I thought adding timeout would help, but it seems to fail with the same error.  Have to find a better way.

Reply:
We have an approach that could be adapted here perhaps.  Just look for CRDs instead of deployments on line 32 [1].

[1] https://github.com/openstack-k8s-operators/install_yamls/blob/main/scripts/operator-wait.sh

Reply:
Thanks! Yeah I was thinking of a while and sleep loop. Though this is hacky would do the job.